lib/pure/net.nim

```diff
@@ -728,6 +728,15 @@ when defineSsl:
     if SSL_set_fd(socket.sslHandle, socket.fd) != 1:
       raiseSSLError()
 
+  proc getAlpnProtocol*(socket: Socket): string =
+    ## Returns the negotiaged ALPN protocols.
+    ## If the ALPN extension is not available or no ALPN candidate
+    ## provided by clients then returns empty string.
+    if not socket.isSsl:
```

Comment:
The `result` is autoinit to `""` then

```nim
if socket.isSsl: result = socket.sslHandle.getAlpnProtocol()
```

Reply:
thank you for your comment, the fix commit is pushed as a405eec1